bubble_sort.rb

```diff
@@ -0,0 +1,44 @@
+# rubocop: disable all
+
+
+def bubble_sort(array_to_sort)
+  array_length = array_to_sort.size
+  return array if array_length <= 1
+
+  # create a loop to loop through array and compare elements
+  loop do
+    swapped = false
+    # subtract one because Ruby arrays are zero-index based
+    (array_length - 1).times do |i|
+      if array_to_sort[i] > array_to_sort[i + 1]
+        array_to_sort[i], array_to_sort[i + 1] = array_to_sort[i + 1], array_to_sort[i]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  puts array_to_sort
+end
+
+def bubble_sort_by(sorted_array)
+  check = true
+  while check
+    check = false
+    (0...sorted_array.length - 1).each do |i|
+      next unless yield(sorted_array[i], sorted_array[i + 1]).positive?
+
+      sorted_array[i], sorted_array[i + 1] = sorted_array[i + 1], sorted_array[i]
+      check = true
+    end
+  end
+  puts sorted_array
```

Comment:
You should return the sorted array, not just print it. Please return the sorted array here.